snakefmt/formatter.py

```diff
@@ -32,7 +33,12 @@
 
 
 def is_all_comments(string):
-    return all(map(comment_start, string.splitlines()))
+    return all(
+        map(
+            comment_start,
+            [s for s in string.splitlines(keepends=True) if s.strip(" \t")],
```

Comment:
could you explain what the `s.strip(" \t")` achieves - and why the conditional `if`?

Reply:
It is basically to ignore "blank" lines (not including newlines). I was very deep in debugging when I did this, but IIRC there was an issue where we weren't detecting a string as being all comments because there was a line of just  tabs/spaces.

Reply:
yeah makes sense!